### PR TITLE
Bookstore with Redux #4: Connect API

### PR DESCRIPTION
**Pull request by miltonHenschel**
- [x] Use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as your backend

- [x] Use either [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) or [axios](https://axios-http.com/docs/intro) with [createAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) to make API calls

- [x] Fetch your books from Bookstore API on when the component has rendered with useEffect

- [x] Refactor your add book and remove book features to persist your changes in the server

- Modify your reducer logic to include HTTP requests to the Bookstore API

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "@testing-library/jest-dom": "^5.16.5",
```
Comment (HAMMAS-SALEEM) [inline package.json:8]: image

- [x] Kindly complete this message such as **If you like this project then don't forget to give a star :star: on this repository** to make your readme file descriptive enough :blush:

Comment (HAMMAS-SALEEM) [inline package.json:8]: image

- [x] Kindly replace this command shown in the image above :arrow_up: with `npm install` to make your readme file descriptive and useful :blush:

**Review by HAMMAS-SALEEM — COMMENTED** [review]
# Invalid Submission :red_circle:

invalid-submission

## Hey @miltonHenschel :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements but I cannot proceed review of your pull request because there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json (hunk 1 above)
Comment (miltonHenschel) [reply package.json:8]: Thanks @HAMMAS-SALEEM . I'll update it.

Comment (miltonHenschel) [reply package.json:8]: Thanks @HAMMAS-SALEEM.

**Review by KDlamini — APPROVED** [review]
### Hi @miltonHenschel 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the project requirements for connecting the Bookstore to an API. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 

✔️  `axios` is used correctly to make the HTTP requests
✔️  `useEffect` is used when dispatching the `fetchBooks` function in `<BookList>`
✔️  `addBook` and `removeBook` methods work as expected
✔️   Data is fetched from API and stored in the Redux store

# Status: Approved ✔️

Great-job-smiley

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**